src/Resolvers/Microsoft.DotNet.NativeWrapper/Interop.cs

```diff
@@ -55,7 +54,36 @@ internal enum hostfxr_resolve_sdk2_result_key_t : int
             global_json_path = 1,
         }
 
-        internal static class Windows
+        [StructLayout(LayoutKind.Sequential, CharSet = CharSet.Auto)]
+        internal struct hostfxr_dotnet_environment_info
+        {
+            public int size;
+            public string hostfxr_version;
+            public string hostfxr_commit_hash;
+            public int sdk_count;
+            public IntPtr sdks;
+            public int framework_count;
+            public IntPtr frameworks;
+        }
+
+        [StructLayout(LayoutKind.Sequential, CharSet = CharSet.Auto)]
+        internal struct hostfxr_dotnet_environment_framework_info
+        {
+            public int size;
```

Comment:
We're discussing if this is actually correct. The declared type in C++ for the size and count fields is `size_t` which to my understanding is a 64bit unsigned integer on 64bit platforms. But C# `int` is always 32bit.
For some reason the structs work though - which is weird. It's as if something is padding the values.
@mateoatr is looking into this.

Reply:
This is working "by accident" - we should fix this. The type should be `nuint`. See the respective fix for the tests https://github.com/dotnet/runtime/pull/49386.

All size and count fields/parameters which are typed as `size_t` should be typed `nuint` in managed.

Reply:
Updated to nuint and merged in the runtime change, thanks!